src/coreclr/vm/stackwalk.cpp

```diff
@@ -3135,6 +3137,12 @@ void StackFrameIterator::PostProcessingForManagedFrames(void)
     m_exInfoWalk.WalkToPosition(GetRegdisplaySP(m_crawl.pRD), (m_flags & POPFRAMES));
 #endif // ELIMINATE_FEF
 
+#ifdef TARGET_X86
+    hdrInfo gcHdrInfo;
+    DecodeGCHdrInfo(m_crawl.codeInfo.GetGCInfoToken(), 0, &gcHdrInfo);
```

Comment:
All other places that decode GCInfo in this file go through `m_crawl.GetCodeManager()`. Is it worth following the pattern?

Reply:
Good idea, but let me do that as a separate cleanup so that this change is just a fix for the bug. I will need to add a new method to the EECodeManager and then I'd use it in the exceptionhandling.cpp too where we currently also have this raw decoding.